src/modules/dayCycle/DayCycle.ts

```diff
@@ -35,12 +36,12 @@ export default class DayCycle {
     });
 
     public static dayCycleMoments: { [dayCycle in DayCyclePart]?: DayCycleMoment } = {
```

Comment:
Is there a reason a Record wasn't used here?
```suggestion
    public static dayCycleMoments: Record<DayCyclePart, DayCycleMoment> = {
```